app/uk/gov/hmrc/hec/util/JsObjectUtils.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.hec.util
+
+import play.api.libs.json.JsObject
+
+object JsObjectUtils {
+
+  implicit class JsObjectOps(private val j: JsObject) extends AnyVal {
+
+    @SuppressWarnings(Array("org.wartremover.warts.Recursion"))
```

Comment:
couldn't quickly spin up a tail recursive version of this function - it's probably possible but the size of our json we'd use this with will mean it's rare that a non-tail recursive implementation will cause an issue